src/MissionParallelCatchup/parallel_catchup_helm/files/logarithmic_range_generator.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/sh
+
+floorSize=$LOGARITHMIC_FLOOR_LEDGERS
+overlapLedgers=$OVERLAP_LEDGERS
+startLedger=$(echo "$STARTING_LEDGER" | awk '{printf "%d", $1}')
+latestLedgerNum=$(echo "$LATEST_LEDGER_NUM" | awk '{printf "%d", $1}')
+numParallelism=$NUM_PARALLELISM
```

Comment:
I know there's only so much defensive coding one can do in a shell script but it helps to check for variables being defined before use and exiting with an error if not.

Reply:
done